### ISSUE TITLE
Feature Request: Add LogLevelOverride during runtime

### ISSUE THREAD

**Issue by JanEggers** [serilog/serilog#1413]
- [x] New Feature

Im trying to close a feature gap to Microsoft.Extensions.Logging which is able to change loglevel (override) configuration on the fly. 

im not sure if its a feature request or a bug but with the current api im not able to add overrides. updating existing overrides works fine.

**Please describe the current behavior?**

```csharp

       [TestMethod]
        public void TestSerilogOverride()
        {
            var nsLogSwitch = new LoggingLevelSwitch(LogEventLevel.Warning);
            var configuration = new LoggerConfiguration()
                .MinimumLevel.Is(LogEventLevel.Warning)
                .MinimumLevel.Override("ns", nsLogSwitch);

            var logger = configuration
                .CreateLogger();
            
            var context = logger.ForContext("SourceContext", "ns.typename");

            Assert.IsFalse(context.IsEnabled(LogEventLevel.Information));
            
            nsLogSwitch.MinimumLevel = LogEventLevel.Information;
            Assert.IsTrue(context.IsEnabled(LogEventLevel.Information));
            nsLogSwitch.MinimumLevel = LogEventLevel.Warning;
            
            
            var typeLogSwitch = new LoggingLevelSwitch(LogEventLevel.Information);
            configuration.MinimumLevel.Override("ns.typename", typeLogSwitch);
            Assert.IsTrue(context.IsEnabled(LogEventLevel.Information));
        }
```

**Please describe the expected behavior?**
Adding new overrides should behave the same as updating an existing override. 

via:
a. add an api "UpdateOverride" to update all existing loggers after a new override is added to the configuration. 

```diff
+ no need to generate pseudo overrides
- there is currently no tracking of "All" loggers so its hard to find out what to update 
```

b. an internal change to
https://github.com/serilog/serilog/blob/8ae332d983b31044f4da0fa34e3b9cb85ba68bc9/src/Serilog/Core/LevelOverrideMap.cs#L67-L81
so that every logger creates a new override if none is existing 

```diff
- increased memory usage for users that dont use overrides
```

**Comment by JanEggers** [comment]
I created a wrapper that manages the log level overrides the same way Microsoft.Extensions.Logging does it. ie lookup a loglevel each time is enabled is checked.

**Comment by nblumhardt** [comment]
Cool, thanks for the follow-up, Jan. I have another idea brewing, I'll drop you a note and let you know if I manage to make any progress with it. Thanks for all of your input on this! 👍

**Comment by JanEggers** [comment]
@nblumhardt FYI my wrapper works fine as long as the default logger / sinks dont restrict log levels.


if I configure the Logger like:

```csharp
new Wrapper(new LoggerConfiguration();
            .WriteTo.Console(Warning)
            .CreateLogger());
```
the warn level filters loglevels my wrapper would accept. Right now the sink is wrapped in 
https://github.com/serilog/serilog/blob/8ae332d983b31044f4da0fa34e3b9cb85ba68bc9/src/Serilog/Configuration/LoggerSinkConfiguration.cs#L77-L87

it would be nice if this switches could be exposed better.

right now I use a override map similar to yours for my wrapper, but that cannot have impact on preconfigured sinks.

I also dont have a suggestion for an api yet.

**Comment by nblumhardt** [comment]
Hi @JanEggers - I found some time to have a shot at this and came up with: https://github.com/nblumhardt/serilog-reload.

It turned out to be very, very tricky to cover that last mile and get overrides applied correctly in all cases. Pretty sure I've got it all covered but ..... still very much in the proof-of-concept stage.

Jumped through a lot of hoops to try to avoid overhead and allocations, so the code is a little sketchy :-) .. Would be great if you end up having a chance to kick the tyres. Cheers!
